lib/nexthop.h

```diff
@@ -206,7 +206,8 @@ uint32_t nexthop_hash_quick(const struct nexthop *nexthop);
 extern bool nexthop_same(const struct nexthop *nh1, const struct nexthop *nh2);
 extern bool nexthop_same_no_labels(const struct nexthop *nh1,
 				   const struct nexthop *nh2);
-extern int nexthop_cmp(const struct nexthop *nh1, const struct nexthop *nh2);
+extern int nexthop_cmp(const struct nexthop *nh1, const struct nexthop *nh2,
+		       bool use_weight);
```

Comment:
maybe it'd be better to add a new api to make this new logic visible, instead of modifying this existing api. that would reduce the footprint of the change - and there's only one place where you want to use the new logic?

Reply:
at the moment.  I am unaware of any other place this needs to be changed.  I can add a new api, it just got a lot messier in the nexthop.c file( which is what I did originally )

Reply:
yeah i'll do it.  The fact that I had that brain fart caused me to already rewrite it like 3 times.
Just annoyed with myself